.github/workflows/validate_tests.yml

```
@@ -0,0 +1,24 @@
+name: Validate test data
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches: '**'
```

Leaving out the branch specifier does the same thing, and is clearer.
```suggestion
  pull_request:
```

---

Adding the `paths` constraint sounds good. I personally find it clearer in this case to be explicit rather than implicit about the `branches` constraint for the `pull_request` trigger. I'm not sure if it's because the `push` trigger does have & need `branches`, or if it is due to habit of seeing it that way for so long.